ksql-engine/src/main/java/io/confluent/ksql/function/UdfCompiler.java

```diff
@@ -136,11 +147,15 @@ public static UdfInvoker compile(final Method method, final ClassLoader loader)
                   + "(args, returnType, metrics);",
               UdfArgSupplier.class, new String[]{"args", "returnType", "metrics"});
 
-      final List<Schema> args = Collections.singletonList(
-          SchemaUtil.getSchemaFromType(valueAndAggregateTypes.left));
+      final Schema argSchema = paramSchema.isEmpty()
+          ? SchemaUtil.getSchemaFromType(valueAndAggregateTypes.left)
+          : LogicalSchemas.fromSqlTypeConverter().fromSqlType(TypeContextUtil.getType(paramSchema));
+      final List<Schema> args = Collections.singletonList(argSchema);
 
-      final Schema returnValue = SchemaUtil.ensureOptional(
-          SchemaUtil.getSchemaFromType(valueAndAggregateTypes.right));
+      final Schema returnValue = returnSchema.isEmpty()
+          ? SchemaUtil.ensureOptional(SchemaUtil.getSchemaFromType(valueAndAggregateTypes.right))
```

Comment:
Not a change from this PR but I'm curious: how come we ensure optionality of the return schema but not the input schema?

Reply:
if the input schema is required, then a `null` value will be treated as an error (e.g. a `null` value will be returned). On the other hand, all schemas must be optional - which is what the output determines.